qt_ui/windows/mission/flight/QFlightCreator.py

```diff
@@ -103,18 +103,23 @@ def __init__(self, game: Game, package: Package, parent=None) -> None:
         # When an off-map spawn overrides the start type to in-flight, we save
         # the selected type into this value. If a non-off-map spawn is selected
         # we restore the previous choice.
-        self.restore_start_type: Optional[str] = None
+        self.restore_start_type = self.game.settings.default_start_type
         self.start_type = QComboBox()
         for start_type in StartType:
             self.start_type.addItem(start_type.value, start_type)
-        self.start_type.setCurrentText(self.game.settings.default_start_type.value)
+        self.start_type.setCurrentText(self.restore_start_type.value)
         layout.addLayout(
             QLabeledWidget(
                 "Start type:",
                 self.start_type,
                 tooltip="Selects the start type for this flight.",
             )
         )
+        squadron: Squadron = self.squadron_selector.currentData()
+        home_base: ControlPoint = squadron.location
+        if isinstance(home_base, OffMapSpawn):
+            self.start_type.setCurrentText(StartType.IN_FLIGHT.value)
+            self.start_type.setEnabled(False)
```

Comment:
Lock the start type if the window initializes with a squadron that spawns "off map"